controllers/lightspeed/openstacklightspeed_controller.go

```diff
@@ -0,0 +1,256 @@
+	Log := r.GetLogger(ctx)
+
+	instance := &lightspeedv1beta1.OpenStackLightspeed{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			Log.Info("OpenStackLightspeed CR not found")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	helper, err := common_helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		Log,
+	)
+
+	// Save a copy of the conditions so that we can restore the LastTransitionTime
+	// when a condition's state doesn't change.
+	savedConditions := instance.Status.Conditions.DeepCopy()
+
+	// Always patch the instance status when exiting this function so we can persist any changes.
+	defer func() {
+		// Don't update the status, if reconciler Panics
+		if r := recover(); r != nil {
+			Log.Info(fmt.Sprintf("panic during reconcile %v\n", r))
+			panic(r)
+		}
+
+		condition.RestoreLastTransitionTimes(&instance.Status.Conditions, savedConditions)
+		// update the Ready condition based on the sub conditions
+		if instance.Status.Conditions.AllSubConditionIsTrue() {
+			instance.Status.Conditions.MarkTrue(
+				condition.ReadyCondition, condition.ReadyMessage)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
+		}
+
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			return
+		}
+
+	}()
+
+	cl := condition.CreateList(
+		condition.UnknownCondition(
+			lightspeedv1.OpenStackLightspeedReadyCondition,
+			condition.InitReason,
+			lightspeedv1.OpenStackLightspeedReadyInitMessage,
+		),
+	)
+
+	instance.Status.Conditions.Init(&cl)
+	instance.Status.ObservedGeneration = instance.Generation
+
+	if !instance.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, helper, instance)
+	}
+
+	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) {
+		return ctrl.Result{}, nil
+	}
+
+	OLSOperatorInstalled, err := lightspeed.IsOLSOperatorInstalled(ctx, helper)
+	if !OLSOperatorInstalled || err != nil {
+		return ctrl.Result{}, fmt.Errorf("installation of OpenShift LightSpeed not detected: %w", err)
+	}
+
+	// NOTE: We cannot consume the OLSConfig definition directly from the OLS operator's code due to
+	// a conflict in Go versions. When this comment was written, the min. required Go version for
+	// openstack-operator was 1.21 whereas OLS operator required at least Go version 1.23. Once the
+	// Go versions catch up with each other we should consider consuming OLSConfig directly from OLS
+	// operator and updating this code and any subsequent code that consumes this structure.
```

Comment:
using the k8s unstructured client to avoid dependencies is a valid thing to do. Even if we temporarily get our golang version aligned with lightspeed it is worth considering whether directly importing the project for just this struct is worth it.

Reply:
Ack, it makes sense, thanks! Should I get rid of the comment, or is it OK if we keep it here?